src/test/utils.test.ts

```diff
@@ -286,4 +286,52 @@ describe('Utils', () => {
 			should.equal('key @ name', utils.generateReferenceValueForLabel(undefined, new ConfigMapRef(new KeyRef('name', 'key'))));
 		});
 	});
+
+	context('Load/Save of a Secrets file', () => {
+
+		let dummyNonSecretFile: string;
+
+		before( () => {
+			dummyNonSecretFile = path.resolve(__dirname, '../../testFixture', `dummy-secret.yaml`);
+			fs.writeFileSync(dummyNonSecretFile, 'test');
+		});
+
+		after( () => {
+			fs.unlinkSync(dummyNonSecretFile);
+		});
+
+		it('should match the secrets model contents between a save and reload to/from a secret file', async() => {
+			const name: string = 'mysecret';
+			const fpOrig: string = path.resolve(__dirname, '../../testFixture', `${name}.yaml`);
+			const fpTest: string = path.resolve(__dirname, '../../testFixture', `${name}2.yaml`);
+			const yamlDoc:SecretConfig = await utils.loadSecretsFromFile(fpOrig);
+			should.exist(yamlDoc);
+			await utils.saveSecretsToFile(yamlDoc, fpTest);
+			should.exist(utils.validateFileNotExisting(name));
+			const yamlDoc2:SecretConfig = await utils.loadSecretsFromFile(fpTest);
+			should.exist(yamlDoc2);
+			yamlDoc.should.deep.equal(yamlDoc2);
+			fs.unlinkSync(fpTest);
+		});
+
+		it('should persist new added secrets entries and be able to retain them from file', async() => {
+			const name: string = 'mysecret';
+			const fpOrig: string = path.resolve(__dirname, '../../testFixture', `${name}.yaml`);
+			const fpTest: string = path.resolve(__dirname, '../../testFixture', `${name}2.yaml`);
+			const yamlDoc:SecretConfig = await utils.loadSecretsFromFile(fpOrig);
+			should.exist(yamlDoc);
+			utils.setSecretValueForKey(yamlDoc, 'mySecretKey', 'ABC123');
+			await utils.saveSecretsToFile(yamlDoc, fpTest);
+			const yamlDoc2:SecretConfig = await utils.loadSecretsFromFile(fpTest);
+			should.exist(yamlDoc2);
+			should.exist(utils.getSecretValueForKey(yamlDoc2, 'mySecretKey'));
```

Comment:
this line can be removed. The line just after there is a more precise check. In the error message, there will be the information if the value is undefined

Reply:
I guess you mean the line with `			should.exist(yamlDoc2);` ?

Reply:
no, I mean `should.exist(utils.getSecretValueForKey(yamlDoc2, 'mySecretKey'));`

this line and the line after is checking both for `utils.getSecretValueForKey(yamlDoc2, 'mySecretKey')`

the line `should.exist(yamlDoc2);` is providing more details in case of test failure. (to know if it that the file wasn't loaded or if it is the values inside it that is not present/cannot be loaded/cannot be read correctly)

Reply:
makes sense